### DISCUSSION TITLE
pull stream from edge

### DISCUSSION THREAD

**Discussion by Bistiara** [OvenMediaLabs/OvenMediaEngine#1723] [Help]
Good day,
I have the following setup: an OME cluster with an origin server, an edge, and Redis as the origin map store. My question is, is it possible to add another edge node, but have it connect to the first edge instead of the origin? The setup would look like this: origin - origin-map-store - edge - edge.

I’ve tried implementing this, but I keep encountering errors.

`ome-edge1  | [2024-10-16 09:42:11.988] E [SPOvtPub-t9000:19] Orchestrator | orchestrator.cpp:740  | Could not find Origin for the stream: [#default#app/stream_test_medium]`

`ome-edge2  | [2024-10-16 09:37:41.382] E [SPRtcSig-t3334:12] OvtStream | ovt_stream.cpp:286  | Describe : Server Failure : 404 (There is no such stream (#default#app/stream_test_medium))`

but if you watch the stream directly from ome-edge1 (app/stream_test_medium), it plays without problems

**Comment by bchah** [comment]
I don't think you can do Edge-Edge but you could use Push Publishing or Stream Map to push from one edge to another origin: https://airensoft.gitbook.io/ovenmediaengine/push-publishing

**Comment by getroot** [comment]
As @bchah  said, when using OriginMapStore, you can only use Origin-Edge structure. Because it uses "app/stream" as key.

Instead, when using OriginMap (https://airensoft.gitbook.io/ovenmediaengine/origin-edge-clustering#originmap), you can use Origin-Relay-Relay-Relay-Edge structure as much as you want. "Relay" requires both OVT Provider and Publisher to be open.
